packages/@react-aria/test-utils/src/dialog.ts

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {act, waitFor, within} from '@testing-library/react';
+import {DialogTesterOpts, UserOpts} from './types';
+
+interface DialogOpenOpts {
+  /**
+   * What interaction type to use when opening the dialog. Defaults to the interaction type set on the tester.
+   */
+  interactionType?: UserOpts['interactionType']
+}
+
+export class DialogTester {
+  private user;
+  private _interactionType: UserOpts['interactionType'];
+  private _trigger: HTMLElement | undefined;
+  private _dialog: HTMLElement | undefined;
+  // TODO: may not need this? Isn't really useful
+  private _dialogType: DialogTesterOpts['dialogType'];
+  private _overlayType: DialogTesterOpts['overlayType'];
+
+  constructor(opts: DialogTesterOpts) {
+    let {root, user, interactionType, dialogType, overlayType} = opts;
+    this.user = user;
+    this._interactionType = interactionType || 'mouse';
+    this._dialogType = dialogType || 'dialog';
+    this._overlayType = overlayType || 'modal';
+
+    // Handle case where element provided is a wrapper of the trigger button
+    let trigger = within(root).queryByRole('button');
+    if (trigger) {
+      this._trigger = trigger;
+    } else {
+      this._trigger = root;
+    }
+  }
+
+  /**
+   * Set the interaction type used by the dialog tester.
+   */
+  setInteractionType(type: UserOpts['interactionType']): void {
+    this._interactionType = type;
+  }
+
+  /**
+   * Opens the dialog. Defaults to using the interaction type set on the dialog tester.
+   */
+  async open(opts: DialogOpenOpts = {}): Promise<void> {
+    let {
+      interactionType = this._interactionType
+    } = opts;
+    let trigger = this.trigger;
+    if (!trigger.hasAttribute('disabled')) {
+      if (interactionType === 'mouse' || interactionType === 'touch') {
+        if (interactionType === 'mouse') {
+          await this.user.click(trigger);
+        } else {
+          await this.user.pointer({target: trigger, keys: '[TouchA]'});
+        }
+      } else if (interactionType === 'keyboard') {
+        act(() => trigger.focus());
+        await this.user.keyboard('[Enter]');
+      }
```

Comment:
```suggestion
    if (!trigger.hasAttribute('disabled')) {
      if (interactionType === 'mouse') {
        await this.user.click(trigger);
      } else if (interactionType === 'touch') {
        await this.user.pointer({target: trigger, keys: '[TouchA]'});
      } else if (interactionType === 'keyboard') {
        act(() => trigger.focus());
        await this.user.keyboard('[Enter]');
      }
```

similar issue with trigger.focus that i brought up above